lib/ansible/modules/windows/win_domain_user.ps1

```diff
@@ -104,12 +207,32 @@ If ($state -eq 'present') {
     }
 
     # Set the password if required
-    If ($password -and (($new_user -and $update_password -eq "on_create") -or $update_password -eq "always")) {
-        $secure_password = ConvertTo-SecureString $password -AsPlainText -Force
-        Set-ADAccountPassword -Identity $username -Reset:$true -Confirm:$false -NewPassword $secure_password -WhatIf:$check_mode @extra_args
-        $user_obj = Get-ADUser -Identity $username -Properties * @extra_args
-        $result.password_updated = $true
-        $result.changed = $true
+    If (
+        (
+            $password
+        ) -and (
+            (
+                $new_user -and $update_password -eq "on_create"
+            ) -or (
+                $update_password -in @("always"; "when_changed")
+            )
+        )
+    ) {
+        If (-not $new_user) { # Don't uncecessary check if the credentials works, if we know it doesn't.
+            $test_new_credentials = Test-Credential -Username $username -Password $password
+        } else {
+            $test_new_credentials = $false
+        }
+        If ($test_new_credentials -and ($update_password -ne "always")) {
+            $result.password_updated = $false
+            $result.changed = $false
```

Comment:
You shouldn't set this to `$false` as it could overwrite code above that set it to `$true`. Just rely on an implicit value of no changes and only set to `$true` if a change occurred.